### ISSUE TITLE
Fatal error when using runProcessor()

### ISSUE THREAD

**Issue by wuuti** [modxcms/revolution#15001]
### Summary
When I tried to use run-processor in an extra connector with 3.x, it fails with a fatal error. The possible reason is that the error handler is never added to the services container, but checked for in the runprocessor method.

### Step to reproduce
Try to call this in a connector:
```php
$response = $this->modx->runProcessor( 'workspace/packages/getlist', $scriptProperties );
```       

### Observed behavior
The following fatal error is thrown and logged:
```log
[Sat Feb 08 10:41:38.893028 2020] [proxy_fcgi:error] [pid 13097] AH01071: Got error 'PHP message: 
PHP Fatal error:  Uncaught MODX\\Revolution\\Services\\NotFoundException: 
Dependency not found with key error. in /vagrant/www/core/src/Revolution/Services/Container.php:28\n
Stack trace:\n
#0 /vagrant/www/core/src/Revolution/modX.php(1733): MODX\\Revolution\\Services\\Container->get('error')\n
#1 /vagrant/www/core/components/updater/model/updater.class.php(395): MODX\\Revolution\\modX->runProcessor('workspace/packages/getlist', Array)\n
#2 /vagrant/www/assets/components/mycomponent/connectors/myconnector.refresh.php(28): Class->myMethod()\n
#3 {main}\n  thrown in /vagrant/www/core/src/Revolution/Services/Container.php on line 28'
```

### Expected behavior
That error should not been thrown, and the error "service" should either be not checked in the runprocessor _or_ initialize it itself if its not there instead of just throwing the exception.

### Possible solution
This can by bypassed if something, e.g. the error handler is added to the container in modX.php:
```php
/**
     * Loads the error handler for this instance.
     *
     * @param array|null $options An array of options for the errorHandler.
     */
    protected function _initErrorHandler($options = null) {
        if ($this->errorHandler == null || !is_object($this->errorHandler)) {
            if ($ehClass = $this->getOption('error_handler_class', $options, modErrorHandler::class, true)) {
                $ehPath = $this->getOption('error_handler_path', $options, '', true);
                if ($ehClass = $this->loadClass($ehClass, $ehPath, false, true)) {
                    if ($this->errorHandler = new $ehClass($this)) {
                        $result = set_error_handler([$this->errorHandler, 'handleError'], $this->getOption('error_handler_types', $options, error_reporting(), true));
                        if ($result === false) {
                            $this->log(modX::LOG_LEVEL_ERROR, 'Could not set error handler.  Make sure your class has a function called handleError(). Result: ' . print_r($result, true));
                        }
                    }
                }
            }
        }
        $this->services->add('error', $this->errorHandler);
    }
```
But I currently don't know if this is the desired service which is checked in the runProcessor method with 
```php
public function runProcessor($action = '', $scriptProperties = [], $options = []) {
        $result = null;

        // Make sure the required services are loaded before initialising a processor
        if (!$this->lexicon) {
            $this->services->get('lexicon');
        }
        if (!$this->error) {
            $this->services->get('error');
        }
...
```
The comment above the check lets me guess that this is kind of unfinished code here. What is the benefit of simply doing a (void) get on the container here without handling any error possibly being thrown? 

### Environment
MODX version 3.0.0-alpha2

**Comment by JoshuaLuckers** [comment]
Related PR: #14920

**Comment by JoshuaLuckers** [comment]
The service with identifier *error* you add to the container in `_initErrorHandler` is an instance of `MODX\Revolution\Error\modErrorHandler`. 
However, a service with identifier *error* is expected to return an instance of  `MODX\Revolution\Error\modError`. 

I believe the right identifier for a service that returns  an instance of `MODX\Revolution\Error\modErrorHandler` is  *error_handler*.

**Comment by Ibochkarev** [comment]
@JoshuaLuckers did you defeat it? Relevant?

**Comment by opengeek** [comment]
Fixed by #15163
